pod-operation/src/components/pressure_transducer.rs

```diff
@@ -2,51 +2,97 @@ use ina219::INA219;
 use rppal::i2c::I2c;
 use tracing::debug;
 
-pub struct PressureTransducer {
-	ina: INA219<I2c>,
-}
-
 // The calibration value is used to adjust the maximum current measurement
 // and precision of measurements.
 const INA219_CALIBRATION_VALUE: u16 = 0xffff;
 
+// The pod will be using two INA219s, so we'll need to differentiate them
+// via their device address.
+const INA219_UPSTREAM_ADDRESS: u8 = 0x40;
+const INA219_DOWNSTREAM_ADDRESS: u8 = 0x41;
+
 // Even with the calibration values, the readings from the INA219 are not in
 // mA. A scaling factor is needed in order to convert the raw reading to mA and
 // this is not provided in the INA219 library that we are using. Note that this
 // value changes according to the calibration value. The exact formula can be
 // found in the INA219 datasheet.
 const INA219_SCALING_VALUE: f32 = 160.0;
 
-// The pressure transducer outputs a current between 4 mA and 20 mA with 0 PSI
-// and 300 PSI respectively. Assuming a linear interpolation, a 1 mA increase
-// results in a 18.75 PSI increase.
-const REF_CURRENT_LOW: f32 = 4.0;
-const REF_CURRENT_HIGH: f32 = 20.0;
-const REF_PRESSURE_LOW: f32 = 0.0;
-const REF_PRESSURE_HIGH: f32 = 300.0;
+struct Reference {
+	pressure_lo: f32,
+	pressure_span: f32,
+	current_lo: f32,
+	current_span: f32,
+}
 
-const REF_CURRENT_SPAN: f32 = REF_CURRENT_HIGH - REF_CURRENT_LOW;
-const REF_PRESSURE_SPAN: f32 = REF_PRESSURE_HIGH - REF_PRESSURE_LOW;
+impl Reference {
+	fn new(pressure_lo: f32, pressure_hi: f32, current_lo: f32, current_hi: f32) -> Self {
+		Self {
+			pressure_lo,
+			current_lo,
+			pressure_span: pressure_hi - pressure_lo,
+			current_span: current_hi - current_lo,
+		}
+	}
+}
```

Comment:
Praise: nice thinking to have this constructor handle the span calculation for us.